library/Zend/Paginator/Paginator.php

```diff
@@ -372,11 +372,8 @@ public function clearPageItemCache($pageNumber = null)
             $cacheIterator = static::$cache->getIterator();
             $cacheIterator->setMode(CacheIterator::CURRENT_AS_KEY);
             foreach ($cacheIterator as $key) {
-                $tags = static::$cache->getTags($key);
-                if ($tags && in_array($this->_getCacheInternalId(), $tags)) {
-                    if (substr($key, 0, $prefixLength) == self::CACHE_TAG_PREFIX) {
-                        static::$cache->removeItem($this->_getCacheId((int)substr($key, $prefixLength)));
```

Comment:
From reading the code, this ensures that the page number is properly injected into the key.